hardware/arduino/avr/cores/arduino/USBCore.cpp

```diff
@@ -317,19 +306,19 @@ int USB_Send(u8 ep, const void* d, int len)
 	return r;
 }
 
-extern const u8 _initEndpoints[] PROGMEM;
-const u8 _initEndpoints[] = 
+u8 _initEndpoints[] =
 {
 	0,
 	
-#ifdef CDC_ENABLED
 	EP_TYPE_INTERRUPT_IN,		// CDC_ENDPOINT_ACM
 	EP_TYPE_BULK_OUT,			// CDC_ENDPOINT_OUT
 	EP_TYPE_BULK_IN,			// CDC_ENDPOINT_IN
-#endif
 
-#ifdef HID_ENABLED
-	EP_TYPE_INTERRUPT_IN		// HID_ENDPOINT_INT
+#ifdef PLUGGABLE_USB_ENABLED
+	//allocate 3 endpoints and remove const so they can be changed by the user
+	0,
+	0,
+	0,
 #endif
 };
 
```

Comment:
USB Banks can have 8,16,32,64 bytes of data. Double or single sized. Double means it has 2x64 in this case. But the u2 Series doesnt support that if you do this on any endpoint. It has less ram for the USB stuff (has a special name in the datasheet). Also to avoid problems with custom endpoint descriptors I'd let the programmer choose if its double banked or not and if it 8,16,32,64 bytes.